### PR TITLE
Roxie suppress logging

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: roxie/ccd/ccdserver.cpp [hunk 1]
```diff
@@ -1316,6 +1316,38 @@ class CRoxieServerActivity : implements CInterfaceOf<IRoxieServerActivity>, impl
...
+    mutable const char *lastLogFormat = nullptr;
+    mutable unsigned logSuppressed = 0;
+    mutable CriticalSection logCrit;
+
+    virtual void CTXLOGva(const LogMsgCategory & cat, const LogMsgJobInfo & job, LogMsgCode code, const char *format, va_list args) const override  __attribute__((format(printf,5,0))) 
+    {
+        StringBuffer text, prefix;
+        text.valist_appendf(format, args);
+        unsigned now = msTick();
+        CriticalBlock b(logCrit);
```
Comment (mckellyln) [inline roxie/ccd/ccdserver.cpp:1331]: Do we need to hold the crit during the actual printing/logging ?  Does it matter ??

**Review by mckellyln — COMMENTED** [review]
Looks very good, just one minor ques about crit.

File: roxie/ccd/ccd.hpp [hunk 2]
```diff
@@ -624,9 +638,26 @@ class ContextLogger : implements IRoxieContextLogger, public CInterface
     virtual void CTXLOGva(const LogMsgCategory & cat, const LogMsgJobInfo & job, LogMsgCode code, const char *format, va_list args) const override  __attribute__((format(printf,5,0))) 
     {
         StringBuffer text, prefix;
-        getLogPrefix(prefix);
         text.valist_appendf(format, args);
+        unsigned now = msTick();
+        CriticalBlock b(crit);
+        if (format == lastFormat && now-lastLogTime < minLogInterval && strsame(text, lastText))
```
Comment (ghalliday) [inline roxie/ccd/ccd.hpp:644]: Does this need to check format (and is it safe)?  If it is an efficiency check there should be a comment - with a comment about requiring constant formats.

Comment (ghalliday) [reply roxie/ccd/ccd.hpp:644]: Should this be a feature of the logging manager instead?

File: roxie/ccd/ccd.hpp [hunk 3]
```diff
@@ -580,6 +580,14 @@ class ContextLogger : implements IRoxieContextLogger, public CInterface
     unsigned ctxTraceLevel;
     mutable CRuntimeStatisticCollection stats;
     unsigned channel;
+
+    // Variables for detecting excessive repeated logging
+    mutable unsigned suppressed = 0;
+    unsigned minLogInterval = 1000;
```
Comment (ghalliday) [inline roxie/ccd/ccd.hpp:586]: Should be configurable, and possibly higher?

**Review by ghalliday — COMMENTED** [review]
A few questions/comments.  Also needs rebasing.

**Comment by richardkchapman** [comment]
Not convinced this is worth doing

**Pull request closed without merging**